version: '3'

services:

  api:
    build: .
    ports:
      - 3003:3000
      - 9229:9229
      - 5555:5555
    volumes:
      - .:/home/node/app
    depends_on:
      apidb:
        condition: service_healthy
      shadowdb:
        condition: service_healthy
    networks:
      myservices:

  apidb:
    image: postgres:15
    volumes:
      - ./.docker/postgresdata:/var/lib/postgresql/data
    environment:
      - POSTGRES_USER=root
      - POSTGRES_PASSWORD=root
      - POSTGRES_DB=api
    ports:
      - "5433:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U root"]
      interval: 5s
      timeout: 10s
      retries: 5
    networks:
      myservices:

  shadowdb:
    image: postgres:15
    volumes:
      - ./.docker/shadowdata:/var/lib/postgresql/data
    environment:
      - POSTGRES_USER=root
      - POSTGRES_PASSWORD=root
      - POSTGRES_DB=api_shadow
    ports:
      - "5434:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U root"]
      interval: 5s
      timeout: 10s
      retries: 5
    networks:
      myservices:

networks:
  myservices:
    driver: bridge
